src/components/fileStorage/uc/deleteUserData.uc.js

```diff
@@ -1,7 +1,81 @@
-const { removePermissionsThatUserCanAccess } = require('./applicationInternal/removePermissions');
+const CryptoJS = require('crypto-js');
+const { Configuration } = require('@hpi-schul-cloud/commons');
 
-const deleteUserData = [removePermissionsThatUserCanAccess];
+const fileStorageProviderRepo = require('../repo/fileStorageProvider.repo');
+const filesRepo = require('../repo/files.repo');
+
+const { facadeLocator } = require('../../../utils/facadeLocator');
+
+const { alert } = require('../../../logger');
+
+const { trashBinResult } = require('../../helper/uc.helper');
+
+const storageBucketName = (schoolId) => `bucket-${schoolId}`;
+
+const decryptAccessKey = async (secretAccessKey) => {
+	const S3_KEY = Configuration.get('S3_KEY');
+	return CryptoJS.AES.decrypt(secretAccessKey, S3_KEY).toString(CryptoJS.enc.Utf8);
+};
+
+const getStorageProvider = async (storageProviderId) => {
+	const storageProvider = await fileStorageProviderRepo.getStorageProviderMetaInformation(storageProviderId);
+	if (storageProvider) storageProvider.secretAccessKey = await decryptAccessKey(storageProvider.secretAccessKey);
+	return storageProvider;
+};
+
+/**
+ * Delete file connections for files shared with user
+ * @param {BSON|BSONString} userId
+ */
+const removePermissionsThatUserCanAccess = async (userId) => {
+	const data = await filesRepo.getFilesWithUserPermissionsByUserId(userId);
+	const complete = await filesRepo.removeFilePermissionsByUserId(userId);
+
+	return trashBinResult({ scope: 'filePermission', data, complete });
+};
+
+/**
+ * Delete personal files from the given user
+ * @param {BSON|BSONString} userId
+ */
+const removePersonalFiles = async (userId) => {
+	const personalFiles = await filesRepo.getPersonalFilesByUserId(userId);
+	const trashBinData = {
+		scope: 'files',
+		data: personalFiles,
+	};
+	if (personalFiles.length === 0) {
+		return { trashBinData, complete: true };
+	}
+	const storageFileNames = personalFiles.map((file) => file.storageFileName);
+
+	const userFacade = facadeLocator.facade('/users/v2');
+	const schoolId = await userFacade.getSchoolIdOfUser(userId);
+
+	const schoolFacade = facadeLocator.facade('school/v2');
+	const school = await schoolFacade.getSchool(schoolId);
+
+	const storageProvider = await getStorageProvider(school.storageProvider);
```

Comment:
New commit in arc chapter is that get -> return null is an error case and should throw